### PR TITLE
Refactor survey cto dag

### LINKED ISSUE #10: Review DAG: pull_survey_cto_data.py
We would like to review and refactor the code for the following dag: https://github.com/hikaya-io/connectors/blob/master/DAGs/pull_survey_cto_data.py

- Review DAGs
- Refactor the DAG
- Suggest any changes to optimize the DAG

### PR DESCRIPTION
**Pull request by JumaKahiga**
#### Goal
Review, refactor, and optimize the `dots_survey_cto_data_pipeline` DAG.

#### What I changed
- Ran Python yapf (technically, this runs by default on every file I work on)
- Hard coded the `default_args` since this should be on a per DAG basis.
- Moved the `on_failure_callback` and `on_success_callback` to the `default_args` since they were similar in all the DAG's tasks.
- Created a task_util to generate a `start_date` for daily tasks.
- Others i.e. change to Pythonic code, remove print, etc.

#### Best Practices I followed
1. Stopped the use of Variables in this DAG. Here is what Airflow says about the use of these:

> You should avoid usage of Variables outside an operator’s execute() method or Jinja templates if possible, as Variables create a connection to metadata DB of Airflow to fetch the value, which can slow down parsing and place extra load on the DB
- [Source](https://airflow.apache.org/docs/stable/best-practices.html#variables)

> While your pipeline code definition and most of your constants and variables should be defined in code and stored in source control, it can be useful to have some variables or configuration items accessible and modifiable through the UI.
- [Source](https://airflow.apache.org/docs/stable/concepts.html#variables)

#### What I am not doing in this PR
- I have not touched the logic of the DAG tasks' code because I'd need more context on how it works. Hoping to do this at a later date.

#### Recommended Future Work
A. I propose removing the `on_success_callback` later. Probably after this K8s [issue](https://github.com/hikaya-io/connectors/issues/13) is done. Generally, more alerts cause more noise and humans start to ignore them: better to have fewer alerts that mean something is not working. 

B. Stop the use of Variables in the whole project (see Best Practices section above)

### CODE REVIEW

File: DAGs/pull_survey_cto_data.py [hunk 1]
```diff
@@ -104,37 +107,31 @@ def save_data_to_db(**kwargs):
         api_data = get_form_data(form)
 
         response_data = [
-            DataCleaningUtil.clean_key_field(
-                item,
-                primary_key
-            ) for item in api_data
+            DataCleaningUtil.clean_key_field(item, primary_key)
+            for item in api_data
         ]
```
Comment (TAnas0) [inline DAGs/pull_survey_cto_data.py:112]: @JumaKahiga  Was this rewrite done by the automatic Python formatting tool you used?
If so, could you please revert it? I find the original more readable

File: DAGs/pull_survey_cto_data.py [hunk 2]
```diff
@@ -104,37 +107,31 @@ def save_data_to_db(**kwargs):
...
+        if isinstance(response_data, (list, )) and len(response_data):
 
-            if SURV_DBMS is not None and (
-                    SURV_DBMS.lower() == 'postgres' or
-                    SURV_DBMS.lower().replace(' ', '') == 'postgresdb'
-            ):
+            if SURV_DBMS is not None and (SURV_DBMS.lower() == 'postgres'
+                                          or SURV_DBMS.lower().replace(
+                                              ' ', '') == 'postgresdb'):
                 """
```
Comment (TAnas0) [inline DAGs/pull_survey_cto_data.py:119]: @JumaKahiga same as the comment above :slightly_smiling_face:

**Review by amosnjoroge — COMMENTED** [review]
@JumaKahiga the refactor looks good especially the logging, Please see Anas's comments. In addition to the future work, when we remove the on failure notification (Removing it makes lots of sense) we should also ensure that k8 deployment has failure notification as well.

File: DAGs/pull_survey_cto_data.py (hunk 1 above)
Comment (JumaKahiga) [reply DAGs/pull_survey_cto_data.py:112]: I'd have to disable it first to stop the formatting or use a different IDE, let me check. Curious, do we have a linting tool that we use to enforce PEP-8 standards in this codebase? cc @sannleen @andrewtpham

**Review by TAnas0 — APPROVED** [review]

**Comment by andrewtpham** [comment]
@sannleen can this now be merged?

**Comment by amosnjoroge** [comment]
@andrewtpham yes it can.

**Pull request merged**